lib/matplotlib/collections.py

```diff
@@ -888,6 +888,17 @@ def set_verts(self, verts, closed=True):
 
     set_paths = set_verts
 
+    def set_verts_and_codes(self, verts, codes):
+        if (len(verts) != len(codes)):
+            raise ValueError("'codes' must be a 1D list or array "
+                             "with the same length of 'verts'")
+        self._paths = []
+        for xy, cds in zip(verts, codes):
+            if len(xy):
+                self._paths.append(mpath.Path(xy, cds))
+            else:
+                self._paths.append(mpath.Path(xy))
+
```

Comment:
I caught that after I pushed the last commit.  I have it in my local branch and I'll push it once I have the testcase figured out (trying to find a simplest case than my 40x40 data that reproduces the error)